pkg/util/time_test.go

```diff
@@ -38,3 +38,52 @@ func TestDurationWithJitter(t *testing.T) {
 func TestDurationWithJitter_ZeroInputDuration(t *testing.T) {
 	assert.Equal(t, time.Duration(0), DurationWithJitter(time.Duration(0), 0.5))
 }
+
+func TestParseTime(t *testing.T) {
+	ts, err := time.Parse(time.RFC3339Nano, "2015-06-03T13:21:58.555Z")
+	require.NoError(t, err)
+
+	var tests = []struct {
+		input  string
+		fail   bool
+		result time.Time
+	}{
+		{
+			input: "",
+			fail:  true,
+		}, {
+			input: "abc",
+			fail:  true,
+		}, {
+			input: "30s",
+			fail:  true,
+		}, {
+			input:  "123",
+			result: time.Unix(123, 0),
+		}, {
+			input:  "123.123",
+			result: time.Unix(123, 123000000),
+		}, {
+			input:  "2015-06-03T13:21:58.555Z",
+			result: ts,
```

Comment:
```suggestion
			result: time.Unix(143333771, 855 * time.Millisecond.Nanoseconds()),
```

Same below. Otherwise it's just testing two calls to time.Parse against each other.

Reply:
I copied the exact implementation and tests of Prometheus. Should we keep it 1-1 with Prometheus or change it? I'm open to discuss it.

Reply:
I'd weakly prefer test to be clearer and using suggested version. I don't see small deviation from Prometheus to be an issue here.

Reply:
Suggested version is incorrect. Correct ts is 1433337718 and 555 milliseconds of course. 🤦

Reply:
Done.